gems/sorbet-runtime/lib/types/props/private/setter_factory.rb

```diff
@@ -11,7 +11,7 @@ module SetterFactory
 
       sig do
         params(
-          klass: T.all(Module, T::Props::ClassMethods),
+          klass: T.all(Module, T::Props::Common::ClassMethods),
```

Comment:
Is it possible to structure the code so that this API doesn't change? Stripe's codebase uses `T::Props::ClassMethods` all over the place and I'd love to not have to do a codemod for this PR.

Reply:
I will give it a try, but it's possible that some of these may need to changed to accommodate for `T::ImmutableStruct`.

Reply:
I have tried reflecting on how to not change these signatures, but I couldn't come up with anything that will satisfy all our needs.

These methods should all work for an `ImmutableStruct`, which does not include `T::Props`. At the same time, we want the behaviour of including `T::Props` to remain the same - which means that `ImmutableStruct` can't include `T::Props`.

Let me know if you have a good idea on how to work around this.